frontend/src/components/Search/Search.jsx

```diff
@@ -1,29 +1,142 @@
+                    setShowDropdown(true);
+                }
+    
+                if (response.data.status === "error") {
+                    setError(`${response.code}: ${response.message}`);
+                }
+            } catch (error) {
+                setError(`${error.message}: Please try again later.`);
+            }
+        } else {
+            setSearchInput([]);
+            setShowDropdown(false);
+        }
+    }
+
+    const handleSearchSubmit = async (event) => {
         event.preventDefault();
         navigate(`stocks/${searchInput}`);
-    }
+    };
 
     return (
-        <div className="search-container">
-            <form className="search-bar" onSubmit={handleSearchSubmit}>
-                <input
-                    type="text"
-                    placeholder="Search for a Stock..."
-                    value={searchInput}
-                    onChange={(event) => setSearchInput(event.target.value)}
-                />
-                <button type="submit">
-                    <i className="material-icons">search</i>
-                </button>
-            </form>
+        <div className="search">
+            <div className="search-container">
+                <form className="search-bar" onSubmit={handleSearchSubmit}>
+                    <input
+                        type="text"
+                        placeholder="Search for a Stock..."
+                        value={searchInput}
+                        onChange={handleSearchChange}
+                    />
+                    <button type="submit">
+                        <i className="material-icons">search</i>
+                    </button>
+                </form>
+                {
+                    showDropdown && 
+                    <div className="dropdown">
+                        {
+                            searchResults.map((stock) => {
+                                if (isValidStock(stock)) {
+                                    return (
+                                        <Link key={stock.instrument_name} to={`/search/stocks/${stock.symbol}`} className="stock-link">
+                                            <div className="dropdown-item">
+                                                <p>{stock.instrument_name} ({stock.symbol})</p>
+                                            </div>
+                                        </Link>
+                                    )
+                                }
+                            })
+                        }
+                    </div>
+                }
+            </div>
+            <table className="search-table">
+                <tbody>
+                    <tr>
+                        <th>Logo</th>
+                        <th>Ticker</th>
+                        <th>Name</th>
+                        <th>Sector</th>
+                        <th>Price</th>
+                    </tr>
+                </tbody>
+                {
```

Comment:
an edge case: if db is empty, this component will only have the table headers. you could add some informative text for users in that case.